### DISCUSSION TITLE
Mesh not starting automatically

### DISCUSSION THREAD

**Discussion by ghost** [Ylianst/MeshCentral#5936] [Issues] [answered]
Hi,

The mesh is not starting automatically and I can only run it manually.
I see this error:

root@Latitude-5480:/usr/local/mesh_services/meshagent# sudo systemctl status meshcentral
meshcentral.service - MeshCentral Server
     Loaded: loaded (/etc/systemd/system/meshcentral.service; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2024-03-15 19:37:49 CET; 21min ago
   Main PID: 8359 (node)
      Tasks: 11 (limit: 18775)
     Memory: 22.8M
        CPU: 3min 40.750s
     CGroup: /system.slice/meshcentral.service
             └─8359 /usr/bin/node meshcentral

Mar 15 19:59:44 Latitude-5480 node[8359]:     at ChildProcess.exithandler (child_process.js:308:12)
Mar 15 19:59:44 Latitude-5480 node[8359]:     at ChildProcess.emit (events.js:326:22)
Mar 15 19:59:44 Latitude-5480 node[8359]:     at maybeClose (internal/child_process.js:1022:16)
Mar 15 19:59:44 Latitude-5480 node[8359]:     at Process.ChildProcess._handle.onexit (internal/child_process.js:287:5) {
Mar 15 19:59:44 Latitude-5480 node[8359]:   killed: false,
Mar 15 19:59:44 Latitude-5480 node[8359]:   code: 1,
Mar 15 19:59:44 Latitude-5480 node[8359]:   signal: null,
Mar 15 19:59:44 Latitude-5480 node[8359]:   cmd: '/usr/bin/node /home/quizhead/meshcentral/node_modules/meshcentral/meshcentral --launch 8359'
Mar 15 19:59:44 Latitude-5480 node[8359]: }
Mar 15 19:59:44 Latitude-5480 node[8359]: ERROR: MeshCentral failed with critical error, check mesherrors.txt. Restarting in 5 seconds...

Not sure what is the problem and how to troubleshoot.

**Comment by si458** [comment]
you can get the full logs using `journalctl -u meshcentral.service`
OR
check the `mesherrors.txt` inside `meshcentral-data` to find out whats wrong

**Comment by ghost** [comment]
This the error I see in the mesherrors

/home/quizhead/meshcentral/node_modules/@yetzt/nedb/lib/datastore.js:77
if (err) { throw err; }
^
[Error: EACCES: permission denied, open '/home/quizhead/meshcentral/meshcentral-data/meshcentral-stats.db'] {
errno: -13,
code: 'EACCES',
syscall: 'open',
path: '/home/quizhead/meshcentral/meshcentral-data/meshcentral-stats.db'
}

Tried to reinstall NPM.

Thanks.

**Reply by si458** [reply, marked as answer]
self explanatory really...
it cant open the `meshcentral-stats.db` because of a permissions problem
so check what user/group created the `meshcentral-stats.db` file,
then check ur `meshcentral.service` file for what the user/group is set too,
make sure they both match...

**Reply by ghost** [reply]
How do I check that? Sorry for noob questions

**Reply by si458** [reply]
`ls -lah /home/quizhead/meshcentral/meshcentral-data/`

`cat /etc/systemd/system/meshcentral.service`

**Reply by ghost** [reply]
Thanks it worked.
My user had the permissions and somehow it changed to root.
Maybe during my many tests trying to solve it. Thank you.
